server/controllers/user_controller.js

```diff
@@ -1,61 +1,132 @@
-        const user = User.create(req.body);
-        return res.status(status.RESOURCE_CREATED).send(user);
+  static model() {
+    return new Model('users');
+  }
+
+  // Get all users
+  static retrieveAllUsers = async (req, res) => {
+    try {
+      const rows = await this.model().select('user_id, email, first_name, last_name, is_admin');
+      // If no users are registered
+      if (rows.length === 0) {
+        return res.status(400).json({
+          status: status.NOT_FOUND,
+          error: 'No user found',
+        });
       }
-      return res.status(status.BAD_REQUEST).send({ status: status.BAD_REQUEST, error: `${result.error.details[0].message}` });
-    };
+      // If all users have been retrieved
+      return res.status(200).json({
+        status: status.REQUEST_SUCCEDED,
+        message: 'All users are retrived successfully',
+        data: rows,
 
-    signIn = (req, res) => {
-      // validation of Request payload
-      // using JOI npm
-      const schema = {
-        email: Joi.string().email().required(),
-        password: Joi.required(),
-      };
-      const result = Joi.validate(req.body, schema);
-      if (result.error == null) {
-      // Everything is okay
-        // We fire up User model to login user
-        const user = User.login(req.body);
-        if (user.status === status.REQUEST_SUCCEDED) {
-          res.set('x-auth-token', user.data.token);
-          return res.status(status.REQUEST_SUCCEDED).send(user);
-        }
+      });
+    } catch (e) {
+      // Catch any error if it rises
+      return res.status(status.SERVER_ERROR).json({
+        status: status.SERVER_ERROR,
+        error: 'server error',
+        e,
+      });
+    }
+  }
+
+  //  Signup
+  static signup = async (req, res) => {
+    try {
+      let {
+        first_name,
+        last_name,
+        email,
+        password,
+        is_admin,
+      } = req.body;
+
+      // Generate a unique ID
+      // Check if Email already exist
+      const user = await this.model().select('*', 'email=$1', [email]);
+      if (user[0]) {
+        // Catch any error if it rises
+        return res.status(status.REQUEST_CONFLICT).json({
+          status: status.REQUEST_CONFLICT,
+          error: `${email} already exists`,
 
-        return res.status(status.UNAUTHORIZED).send(user);
+        });
+      }
+      // Hash user password before being stored
+      password = await hashPassword.encryptPassword(password);
+      // console.log(password);
+      const cols = 'first_name, last_name, email, is_admin, password';
+      const sels = `'${first_name}', '${last_name}', '${email}', '${is_admin}', '${password}'`;
+      const rows = await this.model().insert(cols, sels);
+
+
+      let token = generateAuthToken(rows[0].user_id, rows[0].is_admin);
+
+      return res.status(status.RESOURCE_CREATED).json({
+        status: status.RESOURCE_CREATED,
+        message: 'User signed up successfully',
+        data: {
```

Comment:
@victorkarangwa4 , as I said on the group, you should only return the token and nothing else. please cross check the standards I have set and ensure your implementation conforms with it